### ISSUE TITLE
Linter Rule: No autofocus attribute

### ISSUE THREAD

**Issue by NotGrm** [marcoroth/herb#177]
##### Description

Prevent usage of the `autofocus` attribute

##### Rationale

The autofocus attribute is a global attribute that indicates an element should be focused on page load. Autofocus reduces accessibility by moving users to an element without warning and context. Its use should be limited to form fields that serve as the main purpose of the page, such as the search input on the Google home page.

##### ✅ Good

```html+erb
<input type="text" />
```

##### 🚫 Bad

```html+erb
<input type="text" autofocus />
```

```html+erb
<input type="password" autofocus="autofocus" />
```

#### References

Inspired by [ember-template-lint](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-autofocus-attribute.md)

**Comment by brunoprietog** [comment]
Well, I don't quite agree with this rule. You can and should use autofocus on a popover that has focusable controls, such as a menu. It's also good practice in SPAs to bring the first h1 title to the user when navigating, since screen readers mostly don't report a navigation only when you replace the body element.

I personally use this attribute quite a lot exactly to improve accessibility.

**Comment by natematykiewicz** [comment]
I use it quite a bit to improve accessibility. For example, our OTP page (whose entire purpose is to enter a 6-digit number) autofocuses you to the OTP textbox, so screen users don't have to fumble tabbing through all sorts of UI just to find the textbox.
